### PR TITLE
update cert installation process

### PR DESCRIPTION
**Pull request by maliroteh-sf**
### What does this PR do?
Update cert installation process + make it more robust on Windows platform.

### CODE REVIEW

File: src/common/AndroidTypes.ts
Comment (maliroteh-sf) [inline src/common/AndroidTypes.ts:151]: These are not needed anymore since they've been refactored into `AndroidDevice` and `AndroidDeviceManager`

File: src/common/AndroidUtils.ts
Comment (maliroteh-sf) [inline src/common/AndroidUtils.ts:181]: These are not needed anymore since they've been refactored into `AndroidDevice` and `AndroidDeviceManager`

File: src/common/device/AndroidDevice.ts
Comment (maliroteh-sf) [inline src/common/device/AndroidDevice.ts:23]: Leave it to Google to mess this up. When querying AVDs on different platforms, some of these come back with _ and some with - as work separators. To I updated the logic during parsing to just replace all _ and - with blank space and then we can do a consistent string check on different platforms and for different OS types

File: test/unit/common/AndroidMockData.ts
Comment (maliroteh-sf) [inline test/unit/common/AndroidMockData.ts:9]: updating mock data for more complete testing

File: src/common/AndroidUtils.ts
Comment (maliroteh-sf) [inline src/common/AndroidUtils.ts:504]: I'm making a change to run ALL `adb` commands with retry mechanism. I don't know why we didn't do this in the first place along time ago. It's just a good thing to do since on Windows `adb` barf a lot and we need to run a command again for it to pass.

So by default an `adb` command now will be retried. If the caller really doesn't want the retry mechanism then they can pass in 0 for `numRetries`

File: src/common/device/AndroidDevice.ts
Comment (khawkins) [inline src/common/device/AndroidDevice.ts:33]: Where would we end up using "preferred" vs. "mandatory", out of curiosity?

File: src/common/device/AndroidDevice.ts
Comment (khawkins) [inline src/common/device/AndroidDevice.ts:313]: Nit: I would suggest pulling this boolean comparison out to a more meaningful name for this context. E.g.:

```typescript
const emulatorAlreadyBooted = (resolvedPortNumber === port);
if (emulatorAlreadyBooted) { // ...
```
Just to add clarity of how what this condition means to the casual glance of the maintainer.

File: src/common/device/AndroidDevice.ts
Comment (khawkins) [inline src/common/device/AndroidDevice.ts:341]: Nit:
```suggestion
        // is especially true on the Windows platform. So instead we spawn the process to launch
```

File: src/common/device/AndroidDevice.ts
Comment (khawkins) [inline src/common/device/AndroidDevice.ts:155]: Nit: We also have this functionality [here](https://github.com/forcedotcom/lwc-dev-mobile-core/blob/2b97e570c2fc8352d81cfc76fcd19826113dc6eb/src/common/AndroidUtils.ts#L723). Perhaps we should define a public method for it in `AndroidUtils`, just to keep it consistent?

**Review by khawkins — APPROVED** [review]
Seems good, worth testing out! 👍

File: src/common/device/AndroidDevice.ts
Comment (maliroteh-sf) [reply src/common/device/AndroidDevice.ts:33]: For local dev preview on mobile, the first step would be to boot the device, then see if it has cert installed or not, and if not then install the cert (but to install the cert we need the device to be booted with write system access).

So the LDP CLI command first attempts to boot a device with `preferred` writable-system. If the device is not already booted then we boot it with writable system right off the bat assuming that there may be a need to install the cert and in order to avoid a forced reboot into writable system later one, in order to save time. Even if the device already has the cert installed, booting it in writable system doesn't cause any issues with local dev previewing.

If the device is already booted (for example by AS) then we just continue and won't force-reboot into writable-system. Then we check to see if cert is installed or not. If it is installed then all is good and we continue to use the already booted device as is. But if it is not installed then we force-reboot into writable system.

So having this "preferred" vs. "mandatory" can help us get to local preview faster under certain conditions (b/c of less device reboot)

**Pull request merged**
